### PR TITLE
AIRFLOW-4748 Print stacktrace for dagbag process exceptions

### PR DESCRIPTION
**Pull request by dossett**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/projects/AIRFLOW/issues/AIRFLOW-4748) issues and references them in the PR title. 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes: provide extra error handling information from failed dag parsing

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by dossett** [comment]
One of the test suites passed, some others failed, I can't tell if the failures are connected to this change or not.

File: airflow/models/dagbag.py [hunk 1]
```diff
@@ -198,7 +199,8 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
                     mods.append(m)
                 except Exception as e:
                     self.log.exception("Failed to import: %s", filepath)
-                    self.import_errors[filepath] = str(e)
+                    self.import_errors[filepath] = 'ERROR MESSAGE [' + str(e) + '] ' \
+                                                   'STACK TRACE [' + traceback.format_exc() + ']'
```
Comment (milton0825) [inline airflow/models/dagbag.py]:
```suggestion
                                                   + 'STACK TRACE [' + traceback.format_exc() + ']'
```

File: airflow/models/dagbag.py [hunk 2]
```diff
@@ -233,7 +235,8 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
                         mods.append(m)
                     except Exception as e:
                         self.log.exception("Failed to import: %s", filepath)
-                        self.import_errors[filepath] = str(e)
+                        self.import_errors[filepath] = 'ERROR MESSAGE [' + str(e) + '] ' \
+                                                       'STACK TRACE [' + traceback.format_exc() + ']'
```
Comment (milton0825) [inline airflow/models/dagbag.py]:
```suggestion
                                                       + 'STACK TRACE [' + traceback.format_exc() + ']'
```

**Comment by dossett** [comment]
Failing unit tests fixed.  Searching for `[fail]` in the travis output was the key.

Thanks for the suggestion @milton0825

**Comment by ashb** [comment]
Do you fancy a bit of a deeper change here? Rather than just making import errors a list of strings, seei if we could make it a list of dicts of `{'error': str(e), 'traceback': traceback.format_exc() }` and then we can improve the rendering of this in the UI. This _may_ turn out to be a much bigger change though, but would I think produce a nicer result.

**Comment by dossett** [comment]
@ashb That is an intriguing idea, but I don't have the bandwidth of a broader fix right now.  This is one of a few Airflow PRs I'm working as a result our 1.10.3 upgrade last week.

**Comment by dossett** [comment]
@ashb Are you ok with this more limited change?

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
@dossett Could you provide a screenshot of what this error looks like currently?

I have a feeling we may need to add `whitespace: pre-wrap` to the CSS somewhere otherwise the stack trace may be unreadable.

**Comment by dossett** [comment]
@ashb The way we use Airflow I don't actually see this in the UI.  We have a service to submit the DAGs which calls into Airflow's code to parse the dag first (I don't recall the exact mechanism), so I only see this in an error log.

**Comment by dossett** [comment]
@ashb I stand corrected, I saw it the next time I opened Airflow, lol.

image

**Comment by ashb** [comment]
Hmmmmm - that is rather large and long. Adding the CSS rule I suggested would make it take up about 3 screens. Is that error any use as it stands to you?

**Comment by dossett** [comment]
@ashb No, this particular error isn't that useful, but other errors have been especially very generic python error messages.  It does seem clunky in the UI like this, I admit, but finding it in a log file is super valuable.

**Comment by pgagnon** [comment]
@ashb @dossett I think this is a useful change but the current implementation perhaps adds a bit too much clutter. I could see situations where you have five of these errors taking an inconvenient amount of space. Perhaps it would be better if the stack trace details were hidden behind some kind of expandable details panel?

**Comment by ashb** [comment]
@pgagnon Yeah, that's what I was thinking with https://github.com/apache/airflow/pull/5386#issuecomment-501337380 -- I just worry that this is going to take up too much space on the screen.

**Comment by pgagnon** [comment]
Also nit: The PR title is missing brackets to encapsulate the JIRA id. I'm not sure if this is an actual requirement but it would be more consistent.

**Comment by ashb** [comment]
Going to close this PR for now as the messages it prints will likely be too long and will make a mess of the UI. Sorry. It'll be a nice feature though

**Pull request closed without merging**